packages/share/android/src/main/java/io/flutter/plugins/share/SharePlugin.java

```diff
@@ -29,12 +30,17 @@ private SharePlugin(Registrar registrar) {
   @Override
   public void onMethodCall(MethodCall call, MethodChannel.Result result) {
     if (call.method.equals("share")) {
-      if (!(call.arguments instanceof String)) {
-        result.error("ARGUMENT_ERROR", "String argument expected", null);
+      if (!(call.arguments instanceof Map)) {
+        result.error("ARGUMENT_ERROR", "Map argument expected", null);
         return;
       }
-      final String text = (String) call.arguments;
-      share(text);
+      final Object text = ((Map) call.arguments).get("text");
+      if (text == null || !(text instanceof String) || ((String) text).isEmpty()) {
+        result.error("ARGUMENT_ERROR", "Non-empty text expected", null);
+        return;
+      }
```

Comment:
They are present on the dart side. Just double checking here. Is it conventional not to?

Reply:
I don't think there is any established convention about not double-checking, though it might seem a bit redundant. My comment was about moving the Java check to inside the Java `share` method.

Reply:
Done. Replaced the various result.error with throws and result.notImplemented